lib/microformats/format_parser.rb

```diff
@@ -347,10 +122,262 @@ def check_for_h_properties
           next if prop_entry['type'].nil?
 
           prop_entry['type'].each do |type|
-            @found_prefixes[type.downcase.split('-').first.to_sym] = true
+            @found_prefixes.add(prefix_from_class(type).to_sym)
+          end
+        end
+      end
+    end
+
+    def imply_name(element)
+      unless @properties['name'].nil? && !@found_prefixes.include?(:e) && !@found_prefixes.include?(:p) && !@found_prefixes.include?(:h) && @children.empty?
+        return nil
+      end
```

Comment:
This can be refactored down to:

```ruby
return unless @properties['name'].nil? && !@found_prefixes.include?(:e) && !@found_prefixes.include?(:p) && !@found_prefixes.include?(:h) && @children.empty?
```

In a future PR, we should take a look at that conditional.